File: src/main/java/com/indiepost/service/LinkService.java
package com.indiepost.service;

import com.indiepost.dto.LinkDto;
import com.indiepost.model.analytics.Link;

import java.util.List;

/**
 * Created by jake on 8/10/17.
 */
public interface LinkService {

    LinkDto save(LinkDto linkDto);

    void update(LinkDto linkDto);

    void deleteById(Long id);

    LinkDto findById(Long id);

    LinkDto findByUid(String uid);

    List<LinkDto> findAll();

    Link dtoToLink(LinkDto linkDto);

    LinkDto linkToDto(Link link);

}

File: src/main/kotlin/com/indiepost/service/LinkService.kt
package com.indiepost.service

import com.indiepost.dto.LinkMetadataBookResponse
import com.indiepost.dto.LinkMetadataFlimResponse
import com.indiepost.dto.LinkMetadataResponse
import com.indiepost.dto.analytics.LinkDto
import com.indiepost.model.analytics.Link

/**
 * Created by jake on 8/10/17.
 */
interface LinkService {

    fun save(linkDto: LinkDto): LinkDto

    fun update(linkDto: LinkDto)

    fun deleteById(id: Long)

    fun findById(id: Long): LinkDto?

    fun findByUid(uid: String): LinkDto?

    fun findAll(): List<LinkDto>

    fun dtoToLink(linkDto: LinkDto): Link

    fun linkToDto(link: Link): LinkDto

    fun searchMovies(text: String, limit: Int): List<LinkMetadataFlimResponse>

    fun searchBooks(text: String, size: Int): List<LinkMetadataBookResponse>

    fun getFromUrl(url: String): List<LinkMetadataResponse>
}
